src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunMainMethodRootProvider.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace ILCompiler
+{
+    /// <summary>
+    /// Computes a compilation root based on the entrypoint of the assembly.
+    /// </summary>
+    public class ReadyToRunMainMethodRootProvider : ICompilationRootProvider
```

Comment:
Nit: might be better to just do it in ReadyToRunRootProvider, unless you can think of a scenario where we would want to do ReadyToRun roots and not the entrypoint or vice versa (I can't think of one).